File: generated-code/cpp/codecraft/FileReadWrite/model/Player.cpp
#include "Player.hpp"

namespace model {

Player::Player(int id, int score, int resource) : id(id), score(score), resource(resource) { }

// Read Player from input stream
Player Player::readFrom(InputStream& stream) {
    int id = stream.readInt();
    int score = stream.readInt();
    int resource = stream.readInt();
    return Player(id, score, resource);
}

// Write Player to output stream
void Player::writeTo(OutputStream& stream) const {
    stream.write(id);
    stream.write(score);
    stream.write(resource);
}

// Get string representation of Player
std::string Player::toString() const {
    std::stringstream ss;
    ss << "Player { ";
    ss << "id: ";
    ss << id;
    ss << ", ";
    ss << "score: ";
    ss << score;
    ss << ", ";
    ss << "resource: ";
    ss << resource;
    ss << " }";
    return ss.str();
}

bool Player::operator ==(const Player& other) const {
    return id == other.id && score == other.score && resource == other.resource;
}

}

size_t std::hash<model::Player>::operator ()(const model::Player& value) const {
    size_t result = 0;
    result ^= std::hash<int>{}(value.id) + 0x9e3779b9 + (result << 6) + (result >> 2);
    result ^= std::hash<int>{}(value.score) + 0x9e3779b9 + (result << 6) + (result >> 2);
    result ^= std::hash<int>{}(value.resource) + 0x9e3779b9 + (result << 6) + (result >> 2);
    return result;
}

File: generated-code/kotlin/codecraft/FileReadWrite/src/main/kotlin/trans_gen_test/model/Player.kt
package trans_gen_test.model

import trans_gen_test.util.StreamUtil

/**
 * Player (strategy, client)
 */
class Player {
    /**
     * Player's ID
     */
    var id: Int
    /**
     * Current score
     */
    var score: Int
    /**
     * Current amount of resource
     */
    var resource: Int

    constructor(id: Int, score: Int, resource: Int) {
        this.id = id
        this.score = score
        this.resource = resource
    }

    /**
     * Write Player to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeInt(stream, id)
        StreamUtil.writeInt(stream, score)
        StreamUtil.writeInt(stream, resource)
    }

    /**
     * Get string representation of Player
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Player { ")
        stringBuilder.append("id: ")
        stringBuilder.append(id)
        stringBuilder.append(", ")
        stringBuilder.append("score: ")
        stringBuilder.append(score)
        stringBuilder.append(", ")
        stringBuilder.append("resource: ")
        stringBuilder.append(resource)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Player from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Player {
            var id: Int
            id = StreamUtil.readInt(stream)
            var score: Int
            score = StreamUtil.readInt(stream)
            var resource: Int
            resource = StreamUtil.readInt(stream)
            return Player(id, score, resource)
        }
    }
}